cmd/juju/commands/ssh_container.go

```diff
@@ -248,9 +250,10 @@ func (c *sshContainer) resolveTarget(target string) (*resolvedTarget, error) {
 	if isMetaV2 {
 		meta := charmInfo.Charm().Meta()
 		if c.container == "" {
-			c.container = "charm"
-		} else if _, ok := meta.Containers[c.container]; !ok {
-			containers := []string{"charm"}
+			c.container = charmContainerName
+		}
+		if _, ok := meta.Containers[c.container]; !ok && c.container != charmContainerName {
+			containers := []string{charmContainerName}
```

Comment:
if c.container != charmContainerName:
  containers := []string{charmContainerName}

How is that correct? Shouldn't it be:
  containers := []string{c.container}
?

At the very least we need a comment of why we have to start with the charm container as the first entry in the containers list.